### DISCUSSION TITLE
Capstone - flask -- tidak bisa menjalankan flask

### DISCUSSION THREAD

**Discussion by adiperdana23** [teamalgoritma/community#44] [Q&A] [answered]
Halo, saya tidak bisa menjalankan flask, dan terdapat command 'ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/Users/adhiperdanaputra/opt/anaconda3/envs/capstone-flask/lib/python3.8/site-packages/markupsafe/__init__.py)'

mohon solusinya, thank you

<img width="1440" alt="Screen Shot 2022-04-10 at 19 31 08" src="https://user-images.githubusercontent.com/100086394/162618266-a67f7dc0-c8a3-434c-a79e-c316650b1239.png">

**Comment by onlyphantom** [comment, marked as answer]
From the [changelog](https://markupsafe.palletsprojects.com/en/2.1.x/changes/#version-2-1-0), `soft_unicode` was removed from the latest versions of `markupsafe`. 

> Remove soft_unicode, which was previously deprecated. Use soft_str instead. [#261](https://github.com/pallets/markupsafe/pull/261)

#### Solution / Solusi
You can uninstall the current version of `markupsafe` and then reinstall an earlier version:

```
pip install markupsafe==2.0.1
```
Should work the same if you use conda:
```
conda install markupsafe==2.0.1
```

**Reply by adiperdana23** [reply]
okay, that's works, thank you
